src/runtime_src/core/pcie/driver/linux/xocl/subdev/mb_scheduler.c

```diff
@@ -3114,7 +3132,8 @@ exec_running_to_complete(struct exec_core *exec)
 		if (xcmd->state == ERT_CMD_STATE_RUNNING)
 			exec_query_cmd(exec, xcmd);
 
-		if (xcmd->state >= ERT_CMD_STATE_COMPLETED) {
```

Comment:
I am not comfortable with this change.

Reply:
I would have more conservative fix for this. Close this PR for now.